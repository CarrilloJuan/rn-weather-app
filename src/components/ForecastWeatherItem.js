import React from 'react';
import {Text, StyleSheet, View} from 'react-native';
import PropTypes from 'prop-types';
import WeatherIcon from './WeatherIcon';
import Colors from '../constants/Colors';

export default function ForecastWeatherDay({
  day = 'Mon',
  icon = '10d',
  temp = '26ยบ/12ยบC',
}) {
  return (
    <View style={styles.container}>
      <View style={{flex: 2}}>
        <Text style={styles.text}>{day}</Text>
      </View>
      <View style={styles.temp}>
        <WeatherIcon icon={icon} style={styles.weatherIcon} />
        <Text style={styles.text}>{temp}</Text>
      </View>
    </View>
  );
}

ForecastWeatherDay.propTypes = {
  day: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: 24,
    alignItems: 'center',
  },
  temp: {
    flex: 2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  text: {
    fontSize: 18,
    color: Colors.black,
  },
  weatherIcon: {
    height: 50,
    width: 50,
  },
});
