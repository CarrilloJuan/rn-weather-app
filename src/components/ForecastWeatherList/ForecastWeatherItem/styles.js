import {StyleSheet} from 'react-native';
import Colors from '../../../constants/Colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: 24,
    alignItems: 'center',
  },
  temp: {
    flex: 2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  text: {
    fontSize: 18,
    color: Colors.black,
  },
  weatherIcon: {
    height: 50,
    width: 50,
  },
});
