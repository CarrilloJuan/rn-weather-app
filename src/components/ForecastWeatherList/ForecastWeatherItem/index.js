import React from 'react';
import {Text, View} from 'react-native';
import PropTypes from 'prop-types';

import WeatherIcon from '../../WeatherIcon';
import styles from './styles';

export default function ForecastWeatherDay({day, icon, temp}) {
  return (
    <View style={styles.container}>
      <View style={{flex: 2}}>
        <Text style={styles.text}>{day}</Text>
      </View>
      <View style={styles.temp}>
        <WeatherIcon icon={icon} style={styles.weatherIcon} />
        <Text style={styles.text}>{temp}ÂºC</Text>
      </View>
    </View>
  );
}

ForecastWeatherDay.propTypes = {
  day: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
};
