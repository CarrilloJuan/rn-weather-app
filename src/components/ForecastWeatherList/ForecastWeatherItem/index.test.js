import React from 'react';
import {shallow} from 'enzyme';
import ForecastWeatherItem from '.';

const setup = (propOverrides) => {
  const props = Object.assign(
    {
      day: 'Mon',
      icon: '10d',
      temp: '26ยบ/12ยบC',
    },
    propOverrides,
  );

  const wrapper = shallow(<ForecastWeatherItem {...props} />);

  return {
    props,
    wrapper,
  };
};

describe('CurrentWeather component', () => {
  it('renders as expected', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
