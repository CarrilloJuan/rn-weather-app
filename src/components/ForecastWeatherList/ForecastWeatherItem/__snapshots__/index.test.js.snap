// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 24,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
        }
      }
    >
      Mon
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <WeatherIcon
      icon="10d"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
        }
      }
    >
      26º/12ºC
      ºC
    </Text>
  </View>
</View>
`;
